lib/Tvm.js

```diff
@@ -316,17 +395,24 @@ class Tvm {
       try {
         await this._validateOWCreds(params.owApihost, params.owNamespace, params.owAuth)
         await this._validateNamespaceInApprovedList(params.owNamespace, params.approvedList)
+        await this._validateServiceBasedDenyList(params.denyListUrl, params.owNamespace, params.expirationDuration)
```

Comment:
I am not a fan of calling this for every tvm endpoint. Because the file that is fetched is cosmosdb specific. + As this is a stop gap solution I would avoid abstracting this logic to all services for now.

Can't we move all the logic to cosmos tvm for now ? And make the new variables (file url and metrics url) COSMOS specific ?

That will make things easier to maintain and avoid dependencies from files, cdn service to the denylist.

Reply:
Just because we call it a stop gap fix, doesn't mean we have to use a temporary implementation. I think moving all logic to cosmos is very specific to a service impl is a bad idea.
The current design makes sure that the design to use a deny list is generic and can easily work for any service. 
If required, we can always revisit and improve the design based on our requirement rather than totally discard the work done..

Reply:
.. to avoid azure-blob and aws-s3 tvm to depend on the denylist fetch file for now.

Reply:
We have kept denyListParam as optional, and its only set for Azure Cosmos impl action. This means the validation is disabled for all other services. See - https://github.com/adobe/aio-tvm/blob/throttling_stopgap/lib/Tvm.js#L110 
We won't  need one more param for the same. We follow the same pattern for metrics url, if its set then only we push metrics.

Reply:
ok great missed that ! all good then